src/materialized/src/bin/materialized/main.rs

```diff
@@ -313,6 +313,34 @@ struct Args {
         value_name = "PATH"
     )]
     tls_key: Option<PathBuf>,
+    /// Specifies the tenant id when authenticating users.
+    #[clap(
+        long,
+        env = "MZ_CLOUD_TENANT",
+        required_if_eq("user-authentication", "mzcloud"),
+        hide = true
+    )]
+    mz_cloud_tenant: Option<String>,
+    /// JWK used to validate JWTs during user authentication as a PEM public
+    /// key. Can optionally be base64 encoded.
+    #[clap(
+        long,
+        env = "MZ_CLOUD_JWK",
+        required_if_eq("user-authentication", "mzcloud"),
+        hide = true
+    )]
+    mz_cloud_jwk: Option<String>,
+    /// The full URL (including path) to the api-token endpoint.
+    #[clap(
+        long,
+        env = "MZ_CLOUD_API_TOKEN_URL",
+        required_if_eq("user-authentication", "mzcloud"),
+        hide = true
+    )]
+    mz_cloud_api_token_url: Option<String>,
+    /// Specifies the user authentication method, if differing from the default.
+    #[clap(long, hide = true, possible_values = &["mzcloud"])]
```

Comment:
@antifuchs There's no env var here, forcing this to be a cli arg, but I just realized that maybe your stuff doesn't do custom args easily. Should it also have an env equivalent like the others?